frontend/src/main/web/lib/components/glossary/ColumnHeader.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import {PureRenderMixin} from 'react/addons';
+import { Icon } from 'zanata-ui';
+
+var ColumnHeader = React.createClass({
+  propTypes: {
+    sort: React.PropTypes.oneOf(['ascending', 'descending', null]),
+    value: React.PropTypes.string.isRequired,
+    allowSort: React.PropTypes.bool.isRequired,
+    field: React.PropTypes.oneOf(['src_content', 'trans_content', 'part_of_speech', 'desc', 'trans_count']),
+    onClickCallback: React.PropTypes.func
+  },
+
+  mixins: [PureRenderMixin],
+
+  getInitialState: function() {
+    return {
+      sort: this.props.sort
+    };
+  },
+
+  _handleOnClick: function (event) {
+    if(this.props.allowSort === true) {
+      var asc = this.state.sort === "ascending" ? "descending" : "ascending";
```

Comment:
Why is this called `asc`? Looks like it would be a boolean that says whether or not it would be ascending, but it is a string that indicates the direction of the sorting. Maybe `sortDirection`.